awx/playbooks/library/indirect_instance_count.py

```diff
@@ -0,0 +1,92 @@
+# (C) 2012, Michael DeHaan, <michael.dehaan@gmail.com>
+# (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+    callback: host_query
+    type: notification
+    short_description: for demo of indirect host data and counting, this produces collection data
+    version_added: historical
+    description:
+      - Saves collection data to artifacts folder
+    requirements:
+     - Whitelist in configuration
+     - Set AWX_ISOLATED_DATA_DIR, AWX will do this
+'''
+
+import os
+import json
+
+from ansible.plugins.callback import CallbackBase
+
+
+# NOTE: in Ansible 1.2 or later general logging is available without
+# this plugin, just set ANSIBLE_LOG_PATH as an environment variable
+# or log_path in the DEFAULTS section of your ansible configuration
+# file.  This callback is an example of per hosts logging for those
+# that want it.
+
+
+# Taken from https://github.com/ansible/ansible/blob/devel/lib/ansible/cli/galaxy.py#L1624
+
+from ansible.cli.galaxy import with_collection_artifacts_manager
+from ansible.release import __version__
+
+from ansible.galaxy.collection import find_existing_collections
```

Comment:
TODO: will have to wrap this in a try-except, not currently a public API. Do simple warning of some type if it doesn't work.